internal/core/analysis_api/handlers/github.go

```diff
@@ -0,0 +1,330 @@
+	jsoniter "github.com/json-iterator/go"
+	"go.uber.org/zap"
+	"gopkg.in/yaml.v2"
+
+	"github.com/panther-labs/panther/api/lambda/analysis"
+	"github.com/panther-labs/panther/api/lambda/analysis/models"
+)
+
+const (
+	// github org and repo containing detection packs
+	githubOwner      = "panther-labs"
+	githubRepo       = "panther-analysis"
+	signingKeyID     = "2f555f7a-636a-41ed-9a6b-c6192bf55810" // TODO: update: this is a test key
+	signingAlgorithm = kms.SigningAlgorithmSpecRsassaPkcs1V15Sha512
+	// cache the detection packs to prevent downloading them multiple times
+	// when updating packs (potentially one at a time) via the UI
+	cacheTimeout = 60 * time.Minute
+	// minimum version that supports packs
+	minimumVersionName = "v1.15.0"
+)
+
+var (
+	// cache packs and detections
+	cacheLastUpdated time.Time
+	cacheVersion     models.Version
+	detectionCache   = make(map[string]*tableItem)
+	packCache        = make(map[string]*packTableItem)
+)
+
+func downloadGithubAsset(id int64) ([]byte, error) {
+	rawAsset, url, err := githubClient.Repositories.DownloadReleaseAsset(context.Background(), githubOwner, githubRepo, id)
+	// download the raw data
+	var body []byte
+	if rawAsset != nil {
+		body, err = ioutil.ReadAll(rawAsset)
+		rawAsset.Close()
+	} else if url != "" {
+		body, err = downloadURL(url)
+	}
+	return body, err
+}
+
+func downloadGithubRelease(version models.Version) (sourceData []byte, signatureData []byte, err error) {
+	// Setup options and client
+	// First, get all of the release data
+	release, _, err := githubClient.Repositories.GetRelease(context.Background(), githubOwner, githubRepo, version.ID)
+	if err != nil {
+		return nil, nil, err
+	}
+	// retrieve the signature file and entire analysis zip
+	for _, asset := range release.Assets {
+		if *asset.Name == "panther-analysis-all.sig" {
+			signatureData, err = downloadGithubAsset(*asset.ID)
+		} else if *asset.Name == "panther-analysis-all.zip" {
+			sourceData, err = downloadGithubAsset(*asset.ID)
+		}
+		if err != nil {
+			// If we failed to download an asset, report and return the error
+			zap.L().Error("Failed to download release asset file from repo",
+				zap.String("sourceFile", *asset.Name),
+				zap.String("repository", githubRepo))
+			return nil, nil, err
+		}
+	}
+	return sourceData, signatureData, nil
+}
+
+func downloadURL(url string) ([]byte, error) {
+	response, err := http.Get(url) // nolint:gosec
+	if err != nil {
+		return nil, fmt.Errorf("failed to GET %s: %v", url, err)
+	}
+	defer response.Body.Close()
+
+	body, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to download %s: %v", url, err)
+	}
+	return body, nil
+}
+
+func downloadValidatePackData(version models.Version) error {
+	sourceData, signatureData, err := downloadGithubRelease(version)
+	if err != nil || sourceData == nil || signatureData == nil {
+		zap.L().Error("GitHub release version download failed", zap.Error(err))
+		return err
```

Comment:
Error might be nil here, if somehow err is not nil and sourceData or signatureData is nil. Might be helpful for debugging purposes to add somethings like this
```suggestion
		zap.L().Error("GitHub release version download failed", zap.Error(err), zap.Bool("sourceData nil", sourceData == nil), zap.Bool("signatureData nil", signatureData==nil))
		return err
```